relevanceai/clusterer/clusterer.py

```diff
@@ -1086,11 +1103,201 @@ def fit_predict(self, X):
+                "condition_value": " ",
+            }
+            for f in vector_fields  # type: ignore
+        ]
+
+        print("Fitting and predicting on all documents")
+        # Here we run subfitting on these documents
+        clustered_docs = self.subfit_predict_documents(
+            vector_fields=vector_fields,
+            filters=filters,
+        )
+
+        # Updating the db
+        # print("Updating the database...")
+        # results = self._update_documents(
+        #     self.dataset_id, clustered_docs, chunksize=10000
+        # )
+        # self.logger.info(results)
+
+        # # Update the centroid collection
+        # self.model.vector_fields = vector_fields
+
+        # self._insert_centroid_documents()
+
+        if verbose:
+            print(
+                "Build your clustering app here: "
+                + f"https://cloud.relevance.ai/dataset/{self.dataset_id}/deploy/recent/cluster"
+            )
+
+    def subpartialfit_predict_update(
+        self,
+        dataset,
+        vector_fields: list,
+        filters: list = None,
+        cluster_ids: list = None,
+        verbose: bool = True,
+    ):
+        """
+
+        Retrieve all the documents.
+
+        Parameters
+        ------------
+
+        dataset: Dataset
+            The dataset to call fit predict update on
+        vector_fields: list
+            The list of vector fields
+        filters: list
+            The list of filters
+
+        Example
+        ----------
+
+        .. code-block::
+
+
+        """
+        # Get data
+
+        self._init_dataset(dataset)
+
+        filters = [] if filters is None else filters
+        cluster_ids = [] if cluster_ids is None else cluster_ids
+        # Loop through each unique cluster ID and run clustering
+        parent_field = self._get_cluster_field_name(self.parent_alias)
+
+        print("Getting unique cluster IDs...")
+        unique_clusters = self.unique_cluster_ids(alias=self.parent_alias)
+
+        for i, unique_cluster in enumerate(tqdm(unique_clusters)):
+            cluster_filters = filters.copy()
+            cluster_filters += [
+                {
+                    "field": parent_field,
+                    "filter_type": "category",
+                    "condition": "==",
+                    "condition_value": unique_cluster,
+                }
+            ]
+            self.partial_fit_predict_update(
+                dataset=self.dataset_id,
+                vector_fields=vector_fields,
+                filters=cluster_filters,
+                verbose=False,
+            )
+
+        if verbose:
+            print(
+                "Build your clustering app here: "
+                + f"https://cloud.relevance.ai/dataset/{self.dataset_id}/deploy/recent/cluster"
+            )
+
+    def subfit_predict_documents(
+        self,
+        vector_fields,
+        filters: list,
+        cluster_ids: list = None,
+        verbose: bool = True,
+    ):
```

Comment:
Missing docstring. Additionally, the function signature should be changed to contain:
```
...
filters: Optional[list] = None,
cluster_ids: Optional[list] = None,
...
```